### PR TITLE
Rebased version of CWG Poll 5: P1787R6 Declarations and where to find them

### LINKED ISSUE #4326: [2020-11 CWG Motion 5] P1787R6 Declarations and where to find them
https://wiki.edg.com/pub/Wg21fall2020/StrawPolls/p1787r6.html

### PR DESCRIPTION
**Pull request by tkoeppe**
Fixes #4326
Fixes cplusplus/papers#533

This is a rebase of #4379, which we retain as-is in order to preserve the comments.

### CODE REVIEW

File: source/templates.tex [hunk 1]
```diff
@@ -116,7 +116,8 @@
 In a function template declaration, the \grammarterm{unqualified-id} of the
 \grammarterm{declarator-id} shall be a name.
 \begin{note}
-In a class template declaration, if the \grammarterm{class-name}
+In a class or variable template declaration, if the
+declared name
```
Comment (cpplearner) [inline source/templates.tex:120]: Per P1787, a *simple-template-id* is not a name any more.

File: source/basic.tex [hunk 2]
```diff
@@ -1396,583 +1355,620 @@
...
+a \grammarterm{conversion-type-id} is looked up in the same fashion
+as the \grammarterm{conversion-function-id} in which it appears.
+If that lookup finds nothing, it undergoes unqualified name lookup;
+in each case, only names
+that denote types or templates whose specializations are types are considered.
+
+\pnum
+In a friend declaration \grammarterm{declarator}
+whose \grammarterm{declarator-id} is a \grammarterm{qualified-id}
+whose lookup context is\iref{basic.lookup.qual} a class or namespace $S$,
```
Comment (opensdh) [inline source/basic.tex:1759]: This reproduces a mistake in cross-reference placement from the paper (sorry):
```suggestion
whose lookup context\iref{basic.lookup.qual} is a class or namespace $S$,
```

Comment (tkoeppe) [reply source/basic.tex:1759]: Ah, sorry, I missed that, will fix after merges!

Comment (tkoeppe) [reply source/basic.tex:1759]: Fixed by 2432acacbce559ac6cd3e2878ff68e3b917484ec, thanks!

**Pull request merged**